### DISCUSSION TITLE
DFTB+ MD time

### DISCUSSION THREAD

**Discussion by achyuttt** [dftbplus/dftbplus#1503] [General]
Hello everyone,

I am running DFTB+ MD simulations for a system having 432 atoms in it. I am using 3rd order parameterization and using the relatively robust eigen solver. The SCC tolerance is 10E-5. I am using the recent 24.1 version of DFTB+ and I am completely new in the field of DFT,DFTB and MD simulations. 

I am using 36 cpu cores for my calculation but I think I am not getting the output in the speed as mentioned in many of the papers for DFTB+ MD. Its taking more than 40 minutes for a single SCC cycle. 

#!/bin/bash
#SBATCH -N 1
#SBATCH --ntasks-per-node=36
#SBATCH -t 5-48:00:00
#SBATCH --mem=256g
## Activate the DFTB+ environment
ulimit -s unlimited
source ~/.bashrc
module load miniconda3/23.9.0-0
conda activate /data/girilab/dftbplus
# Set the number of OpenMP threads
export OMP_NUM_THREADS=1
# Run DFTB+ and log the output
mpirun -np 36 dftb+ | tee output

When I use this bash script, my job was killed continuously displaying OutOfMemory. And I installed "mamba install 'dftbplus=*=mpi_mpich_*'" this version of DFTB+.

Can anyone suggest how can I improve the speed by looking at this scripts ? Am I doing some mistakes here? please let me know

**Comment by aradi** [comment]
Not sure, why you get out of memory messages.

Howevery, yYour system is actually very small (on DFTB-scale), so it will likely not scale up to 36 MPI-processes. Try first one process, measure the time for a few MD-step, then use two, etc. Find out the sweet spot on your system.

**Reply by achyuttt** [reply]
What could be the normal expected time for 1 MD step for a system of 432 atoms? 

***  Geometry step: 0

--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 0 with PID 85279 on node n100 exited on signal 9 (Killed).
--------------------------------------------------------------------------
slurmstepd: error: Detected 1 oom-kill event(s) in step 332369.batch cgroup. Some of your processes may have been killed by the cgroup out-of-memory handler.

I am getting these kind of errors frequently. 

**2nd Problem**
Even simply a Density of States calculation for <300 atoms is taking around half an hour with a mesh of 444 and SCC Tolerance of 10E-5. Does it seem to be reasonable? or am I doing some resource over allocation?

My bash script looks like this.

#!/bin/bash
#SBATCH -N 1                          
#SBATCH --ntasks=64                   
#SBATCH --cpus-per-task=1               
#SBATCH --mem-per-cpu=4G               
#SBATCH -t 0-24:00:00                  
#SBATCH --partition=uri-cpu,cpu,cpu-preempt,cpu-long
#SBATCH --array=1-43%5                   
##SBATCH --job-name=high_throughput      
#SBATCH --output=array_job_%A_%a.out    

# Load necessary modules
module load miniconda/22.11.1-1

# Activate conda environment
conda activate dftbplus

# Set number of threads (if applicable)
export OMP_NUM_THREADS=1
 .
.
.
.
dftb+ | tee output
